### PR TITLE
drafting parameter

### PR DESCRIPTION
**Pull request by mmaelicke**
Some first drafts for filling the Parameter specification.

- [x] The specifications are not yet complete.
- [x] An example is missing

### CODE REVIEW

**Comment by AlexDo1** [comment]
I really like the text you drafted here, the descriptions are very clear and comprehensible.

I just fixed some typos directly in this branch.

**Comment by mmaelicke** [comment]
This one is ready for review

**Comment by mmaelicke** [comment]
@elnazazmi, @MarcusStrobl  you are also welcome to review this core part of tool-specs concerning the input parameters

**Review by AlexDo1 — APPROVED** [review]

File: docs/parameter.md [hunk 1]
```diff
@@ -1,3 +1,170 @@
...
+
+### `optional`
+
+Boolean field which defaults to `false`. If set to `optional=true`, the parameter is not required by the tool. This implies, that the tool implementation can handle a `parameter.json` in which the `Parameter` is entirely missing.
+
+### `default`
+
+The `default` field is of same data type as the `Parameter` itself. If a default value is set, the tool-framework is required to inject this parameter into the `parameters.json`, as the tool will treat the default like any other non-optional parameter. 
+
+## Example
```
[the remark this answers is not in the document]
Comment (mmaelicke) [reply docs/parameter.md:147]: nice, thanks!

**Pull request merged**